pkg/generate/code/check.go

```diff
@@ -132,37 +138,76 @@ func checkRequiredFieldsMissingFromShape(
 			missing = append(missing, primaryARNCondition)
 			continue
 		}
-		cleanMemberNames := names.New(memberName)
-		cleanMemberName := cleanMemberNames.Camel
 
-		resVarPath := koVarName
-		// Check that the field has potentially been renamed
-		renamedName, wasRenamed := r.InputFieldRename(
-			op.Name, memberName,
-		)
-		_, found := r.SpecFields[renamedName]
-		if found && !wasRenamed {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + cleanMemberName
-		} else if found {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + renamedName
-		} else {
-			_, found = r.StatusFields[memberName]
-			if !found {
+		resVarPath, err := getSanitizedMemberPath(memberName, r, op, koVarName)
+		if err != nil {
+			// If it isn't in our spec/status fields, we have a problem!
+			msg := fmt.Sprintf(
+				"GENERATION FAILURE! there's a required field %s in "+
+					"Shape %s that isn't in either the CR's Spec or "+
+					"Status structs!",
+				memberName, shape.ShapeName,
+			)
+			panic(msg)
+		}
+		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
+	}
+
+	// Loop over the not-required member fields in the shape and determine
+	// if an ID field can be populated by an ID in either the Status or the Spec of the resource
+	// and generate an if condition checking for all required fields having non-nil
+	// corresponding resource Spec/Status values (required for ReadMany)
+	for field := range notRequiredFields {
+		crKind := strings.ToLower(r.Kind)
+		if strings.Contains(strings.ToLower(field), crKind) {
```

Comment:
You probably want to check `strings.HasPrefix` here instead of `strings.Contains`. Consider a `crKind` of `User`. Then consider that there might be a field called `MasterUserName` or `DefaultUsers`, neither of which may refer to a User Kind but rather some other user-ish concept.